src/tsd.c

```diff
@@ -360,11 +365,21 @@ _malloc_thread_cleanup(void) {
 			}
 		}
 	} while (again);
+#if defined(JEMALLOC_HAVE_DLSYM) && !defined(_WIN32)
+	void (*libc_thread_cleanup)() = dlsym(RTLD_NEXT,
+	    "_malloc_thread_cleanup");
+	if (libc_thread_cleanup != NULL) {
+		libc_thread_cleanup();
+	}
+#endif
```

Comment:
Discussed offline. IIUC, we are making sure that there is only one `_malloc_tsd_cleanup_register`, just like `_malloc_thread_cleanup`. So if the user jemalloc wins these definitions, the system jemalloc can just register its cleanup callbacks to be handled together by the user jemalloc, is that right?

Also, if we don't want to workaround for older versions, we can avoid the `dlsym` (which would be my preference)?

Reply:
Yes, this is correct understanding. And I've removed `dlsym`.